src/procedures/iotscape/iotscape-services.js

```diff
@@ -257,7 +257,34 @@ IoTScapeServices.call = async function (service, func, id, ...args) {
       }, 3000);
     }),
   ]).then((result) => result).catch(() => {
-    throw new Error("Response timed out.");
+    // Make second attempt
```

Comment:
Refactoring this into something like `retry(fn, maxAttempts)` or something could make it easier to test and reduce code duplication.

Reply:
I refactored it to be a little cleaner.